plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableParameter.java

```diff
@@ -184,10 +190,11 @@ private Object uncoerce(Object o, Type type) {
         } else if (o != null && !o.getClass().getName().startsWith("java.")) {
             try {
                 // Check to see if this can be treated as a data-bound struct.
-                Map<String, Object> nested = DescribableModel.uninstantiate_(o);
+                UninstantiatedDescribable nested = DescribableModel.uninstantiate2_(o);
                 if (type != o.getClass()) {
-                    nested.put(CLAZZ, o.getClass().getSimpleName());
+                    nested.setKlass(o.getClass().getSimpleName());
                 }
+                nested.setSymbol(symbolOf(o));
```

Comment:
I think we should require it even for the homomorphic collection for clarity. It'll also be less work and tolerates the schema evolution better.